Sources/DependencyCalculator/PackageMetadata.swift

```diff
@@ -127,4 +135,24 @@ struct PackageTargetMetadata {
     func targetIdentity() -> TargetIdentity {
         return TargetIdentity.package(path: path, targetName: name, testTarget: testTarget)
     }
+
+    private static func isSwiftVersion6Plus() throws -> Bool {
+        let versionString = try Shell.execOrFail("swift --version")
```

Comment:
Would it make sense to cache the version value once it's received?

Reply:
Thanks, that was an oversight on my part. Packages can actually be multiple, so caching the version output makes sense.
https://github.com/mikeger/XcodeSelectiveTesting/blob/094be230697a4e0f62d01ae32fecf9c242568f21/Sources/DependencyCalculator/DependencyGraph.swift#L203-L205

Reply:
Fixed in https://github.com/mikeger/XcodeSelectiveTesting/pull/38/commits/18fe3eb7278d3dbd48d3af778fab3ed72beed1a1. The only part actually affected was the concurrent parsing process, so I modified it to check in advance if Swift 6 or later is being used and pass a flag accordingly. Since Swift commands are not executed in parallel in tests or other areas, I determined that addressing this part alone would be sufficient. I have confirmed that `swift package xcode-selective-test` succeeds without locking.

Reply:
Thank you